workerpool/example/consumer.go

```diff
@@ -11,44 +13,79 @@ import (
 type consumer struct {
 	log zerolog.Logger
 
+	tid                 workerpool.ThreadID
 	queue               workerpool.SubmissionQueue
 	completed           uint64
+	workRequested       uint64
 	workCompletedReal   uint64
 	workCompletedCanary uint64
+	workErrReal         uint64
+	workErrCanary       uint64
+	rampDuration        time.Duration
```

Comment:
I'm much more interested in a jitter function than a ramp-up period, but this is good.

Reply:
For sure, that's something we can mix in at a later date. For now I primarily wanted to provide an example for potential consumers of where to stick such an implementation. It isn't a huge requirement at the moment.